invoices/invoiceregistry.go

```diff
@@ -897,22 +868,23 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 			// with our peer. Note that the outcome of the
 			// resolution is set based on the outcome of the single
 			// htlc that we just settled, so may not be accurate
-			// for all htlcs.
-			resolution := *NewSettleResolution(
-				invoice.Terms.PaymentPreimage, key,
-				acceptHeight, result,
-			)
-
-			i.notifyHodlSubscribers(resolution)
+			// for all htlcs. We set the circuit key on the
+			// resolution for each htlc, but the remainder of the
+			// resolution is based on the most recently settled
+			// htlc.
+			settleResolution.circuitKey = key
```

Comment:
Isn't the same resolution updated every iteration? Maybe it works because of some copying happening later, but it looks a risky.

Reply:
We can also take the accept height from the invoice htlc here, instead of passing back the accept height of the current htlc.